### PR TITLE
meta: add a release procedure

### PR DESCRIPTION
**Pull request by Geertiebear**
Open to suggestions and comments here for sure.

### CODE REVIEW

File: RELEASE_PROCEDURE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Versioning
+
+Mlibc uses semantic versioning to denote releases.
```
Comment (ArsenArsen) [inline RELEASE_PROCEDURE.md:3]: don't we stylize as `mlibc` (lowercase)?

**Review by ArsenArsen — APPROVED** [review]
lgtm, generally, the only discussion that one might want to have is about the density of the minor releases

File: RELEASE_PROCEDURE.md (hunk 1 above)
Comment (Geertiebear) [reply RELEASE_PROCEDURE.md:3]: Mhm yeah we could

File: RELEASE_PROCEDURE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+A bump in the major version signals an ABI break.
+A bump in the minor version signals a newly released set of features, while still maintaining ABI.
+A bump in the patch signals a bug-fix, while not adding new features.
+
+# Release schedule
+
+A major version will be released irregularly, as it is only necessary when breaking ABI.
+A minor version will be released bi-monthtly (read: every two months).
```
Comment (avdgrinten) [inline RELEASE_PROCEDURE.md:11]: It's turn that into: A minor or major version will be released every two months. We can turn the release into a major version if there is a pending ABI break. Additionally, we can promise not to break ABI more than once every 6 months or so.

We should also specify in *which* months we do the release and whether the releases are scheduled at the beginning or end of these months.

Comment (Geertiebear) [reply RELEASE_PROCEDURE.md]: Fixed.

**Pull request merged**